### ISSUE TITLE
Dynamic nodes Typescript error

### ISSUE THREAD

**Issue by hthomas92** [newcat/baklavajs#246]
Hello,
when trying to create a Dynamic node based on the example [here](https://v2.baklava.tech/nodes/dynamic-nodes.html), I get a Typescript error. The example works though, I was just wondering about how to get rid of the error.

```
Type '({ operation }: { operation: string; }) => { inputs: { input1: () => NumberInterface; input2?: undefined; }; } | { inputs: { input1: () => NumberInterface; input2: () => NumberInterface; }; }' is not assignable to type '(inputs: { operation: string; }, outputs: { result: number; }) => DynamicNodeUpdateResult'.
  Type '{ inputs: { input1: () => NumberInterface; input2?: undefined; }; } | { inputs: { input1: () => NumberInterface; input2: () => NumberInterface; }; }' is not assignable to type 'DynamicNodeUpdateResult'.
    Type '{ inputs: { input1: () => NumberInterface; input2?: undefined; }; }' is not assignable to type 'DynamicNodeUpdateResult'.
      Types of property 'inputs' are incompatible.
        Type '{ input1: () => NumberInterface; input2?: undefined; }' is not assignable to type 'DynamicNodeDefinition'.
          Property 'input2' is incompatible with index signature.
            Type 'undefined' is not assignable to type '() => NodeInterface<any>'.ts(2322)
```

**Comment by newcat** [comment]
Thanks for the issue, this will be fixed in the next release. For now, you can add `as DynamicNodeDefinition` to the return value of the `onUpdate` function, like this:

```ts
onUpdate({ operation }) {
    if (operation === "Sine") {
        return {
            inputs: {
                input1: () => new NumberInterface("Input", 0),
            } as DynamicNodeDefinition,
        };
    } else {
        return {
            inputs: {
                input1: () => new NumberInterface("Input", 0),
                input2: () => new NumberInterface("Input", 0),
            } as DynamicNodeDefinition,
        };
    }
},
```

**Comment by newcat** [comment]
Fixed in [v2.0.2-beta.4](https://github.com/newcat/baklavajs/releases/tag/v2.0.2-beta.4)
